angular/index.d.ts

```diff
@@ -1035,13 +1035,14 @@ declare namespace angular {
          *
          * @param reason Constant, message, exception or an object representing the rejection reason.
          */
-        reject(reason?: any): IPromise<any>;
+        reject(reason?: any): IPromise<never>;
         /**
          * Wraps an object that might be a value or a (3rd party) then-able promise into a $q promise. This is useful when you are dealing with an object that might or might not be a promise, or if the promise comes from a source that can't be trusted.
          *
          * @param value Value or a promise
          */
         resolve<T>(value: IPromise<T>|T): IPromise<T>;
+        resolve<T1, T2>(value: IPromise<T1>|T2): IPromise<T1|T2>;
```

Comment:
Why is the proposed signature better? In `lib.es2015.promise.d.ts`, we see this:
```ts
resolve<T>(value: T | PromiseLike<T>): Promise<T>;
```
See my previous comment about compatibility between different types of promises. Their types should be structurally compatible as they all implement the same Promises/A+ spec. If the proposed signature is more accurate, then probably `lib.es2015.promise.d.ts` should be changed too.